sorc/chgres_cube.fd/program_setup.f90

```diff
@@ -98,16 +94,71 @@ module program_setup
 !                                 These names will be used to identify
 !                                 the tracer records in the output files.
 !                                 Follows the convention in the field table.
+!                                 FOR GRIB2 FILES: Not used. Tracers instead taken
+!                                 from the varmap file. 
 ! tracers_input                   Name of each atmos tracer record in 
 !                                 the input file.  May be different from
 !                                 value in 'tracers'. 
+!                                 FOR GRIB2 FILES: Not used. Tracers instead taken
+!                                 from the varmap file. 
 ! use_thomp_mp_climo              When true, read and process Thompson
 !                                 MP climatological tracers.  False,
 !                                 when 'thomp_mp_climo_file' is NULL.
 ! vcoord_file_target_grid         Vertical coordinate definition file
 ! wltsmc_input/target             Wilting point soil moisture content
 !                                 input/target grids
-! 
+!
+! nsoill_out                      Number of soil levels desired in the output data. 
+!                                 chgres_cube can interpolate from 9 input to 4 output
+!                                 levels. 
+!                                 DEFAULT: 4    
+!
+! Variables that are relevant only for "grib2" input type:
+!
+! grib2_file_input_grid           REQUIRED. File name of grib2 input data.
+!                                 Assumes atmospheric and surface data are in a single 
+!                                 file. 
+!
+! varmap_file                     REQUIRED. Full path of the relevant varmap file. 
+!
+! external_model                  The model that the input data is derived from. Current
+!                                 supported options are: "GFS", "HRRR", "NAM", "RAP". 
+!                                 Default: "GFS"
+!
+! vgtyp_from_climo                If false, interpolate vegetation type from the input 
+!                                 data to the target grid instead of using data from 
+!                                 static data. Use with caution as vegetation categories
+!                                 can vary. 
+!                                 Default: False
+!
+! sotyp_from_climo                If false, interpolate soil type from the input 
+!                                 data to the target grid instead of using data from 
+!                                 static data. Use with caution as the code assumes
+!                                 input soil type use STATSGO soil categories.
+!                                 Default: False
+!
+! vgfrc_from_climo                If false, interpolate vegetation fraction from the input 
```

Comment:
Why set veg fraction from the input file?  The grid generation process already creates that field from 0.144-degree monthly data.  Is that data too coarse?  If you have a better dataset, we should make it an option for the when making a model grid.   Also, the sfcsub.F component of the model will update veg fraction every 24-hours using either the gaussian file or the tiled file created during the grid generation process.  So, if you run with sfcsub.F turned on, your high-res veg fraction will get wiped out after one day.

Reply:
@GeorgeGayno-NOAA We use real-time NOAA VIIRS greenness fraction at 4-km resolution (this data is also in testing at MCEP) for the HRRR, and as I mentioned in the other comment, we need this to directly compare the FV3-LAM to the operational HRRR for readiness testing of the RRFS.  

We need to turn of the sfcsub.F component of the model for the FV3-LAM, based on what you're saying.  What are the namelist entries that control this part of the code?

Reply:
> @GeorgeGayno-NOAA We use real-time NOAA VIIRS greenness fraction at 4-km resolution (this data is also in testing at MCEP) for the HRRR, and as I mentioned in the other comment, we need this to directly compare the FV3-LAM to the operational HRRR for readiness testing of the RRFS.
> 
> We need to turn off the sfcsub.F component of the model for the FV3-LAM, based on what you're saying. What are the namelist entries that control this part of the code?

The vegetation fraction update may be turned off by setting its nudging coefficient as follows:  FVEGL=99999.

Reply:
Thanks, @GeorgeGayno-NOAA.  After some digging, @LarissaReames-NOAA and I ran into this issue previously and have set fhcyc in the FV3-LAM to avoid all surface nudging, since it isn't necessary at mesoscale NWP forecast lengths.